srcdeps-maven-plugin/src/main/java/org/srcdeps/mvn/plugin/SrcdepsInitMojo.java

```diff
@@ -0,0 +1,516 @@
+                projectBuildingRequest.setProcessPlugins(false);
+                projectBuildingRequest.setRepositoryMerging(ProjectBuildingRequest.RepositoryMerging.REQUEST_DOMINANT);
+                projectBuildingRequest.setSystemProperties(session.getSystemProperties());
+                projectBuildingRequest.setRepositorySession(repoSession);
+
+                Artifact pomArtifact = repositorySystem.createProjectArtifact(g, a, v, "compile");
+                try {
+                    ProjectBuildingResult result = projectBuilder.build(pomArtifact, projectBuildingRequest);
+                    MavenProject dependencyProject = result.getProject();
+                    String url = getScmUrl(dependencyProject, null);
+                    if (url == null) {
+                        log.warn(String.format("No SCM connection for artifact [%s:%s:%s]", g, a, v));
+                    } else {
+                        url = url.substring("scm:".length());
+                        if (isScmUrlSupported(url)) {
+                            log.info(String.format("Found SCM URL [%s] for GAV [%s]", url, gav));
+                            this.add(url, gav);
+                        } else {
+                            log.debug(String.format("Unsupported SCM URL [%s] for GAV [%s]", url, gav));
+                        }
+                    }
+                } catch (ProjectBuildingException e) {
+                    final String msg = String.format("Could not resolve [%s] using remote repositories [%s]",
+                            pomArtifact, remoteRepos);
+                    if (failOnUnresolvable) {
+                        throw new MojoExecutionException(msg, e);
+                    } else {
+                        log.warn(msg);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Creates a new {@link ScmRepository.Builder} and sets its selectors, SCM url and id based on the information
+         * available in this {@link ScmRepositoryIndex}.
+         *
+         * @param url
+         *            the SCM URL to create a new {@link ScmRepository.Builder} for
+         * @return a new {@link ScmRepository.Builder}
+         */
+        public ScmRepository.Builder createRepository(String url) {
+
+            ScmRepository.Builder repoBuilder = ScmRepository.builder();
+            Set<String> selectors = new TreeSet<>();
+
+            Map<String, Integer> groupIdCntMap = urlGroupIdMap.get(url);
+            if (groupIdCntMap.size() == 1) {
+                String groupId = groupIdCntMap.keySet().iterator().next();
+                Set<String> urls = groupIdUrlMap.get(groupId);
+
+                if (urls.size() == 1) {
+                    repoBuilder.id(groupId);
+                    selectors.add(groupId);
+                } else {
+                    Set<Gav> gavs = urlGavMap.get(url);
```

Comment:
Let me try to explain it here and re-use the same text in the JavaDoc:

The main responsibility of `SrcdepsInitMojo` is to produce a `srcdeps.yaml` and make it as complete as possible. To accomplish this, the mojo crawls through the dependency graph of the current project tree and collects the following info:

* SCM URLs (via `<scm>` tags in pom.xml files) to build the dependencies from
* Association between GAVs and SCM URLs - i.e. which GAV should be built from which SCM URL

That is basically enough on the level of raw data, but to produce a nice `srcdeps.yaml` file, we need a bit more:  It is often stupid to list per-artifactId selectors [1] like 

```
selectors:
- org.mygroup:my-artifact-1
- org.mygroup:my-artifact-2
- org.mygroup:my-artifact-3
- org.mygroup:my-artifact-4
```

that would be correct, but not nice and reliable because org.mygroup may decide to add my-artifact-5 at some point in the future. If all artifacts for the given URL, have the same groupId, then we could theoretically generalize the selectors to just

```
selectors:
- org.mygroup
```

but to do that, we need to make sure that the same group does not occur under other URLs. Sadly, that is quite common out there in the wild: e.g. the groupId org.jboss.logging is used by three independent git repos:  https://github.com/jboss-logging/jboss-logging https://github.com/jboss-logging/jul-to-slf4j-stub and https://github.com/jboss-logging/jboss-logging-tools 

We need to both identify these situations and solve them somehow. 

Except for solving selector clashes, we also need a unique id for each SCM repository element in `srcdeps.yaml` file. That's quite a similar problem. We need all those tracking maps in `ScmRepositoryIndex` to solve these two problems.

[1] https://github.com/srcdeps/srcdeps-core/blob/master/doc/srcdeps.yaml#L204